Assets/Scripts/Features/FloatingText/FloatingTextVisual.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections;
+using Core;
+using Services;
+using TMPro;
+using UnityEngine;
+
+namespace Game
+{
+    public class FloatingTextVisual : BaseVisual<FloatingText>
+    {
+        [SerializeField] TMP_Text floatingTextPrefab;
+
+        private Coroutine popCoroutine;
+
+        public void TurnOffPrefab()
+        {
+            floatingTextPrefab.gameObject.SetActive(false);
+        }
+
+        public void PopText(Vector3 position, string text, float duration)
+        {
+           popCoroutine = StartCoroutine(PopTextRoutine(position, text, duration));
+        }
+        
+        IEnumerator PopTextRoutine(Vector3 pos, string text, float duration)
+        {
+            var popText = Summoner.CreateAsset(floatingTextPrefab, transform);
+            var worldPos = UnityEngine.Camera.main.WorldToScreenPoint(pos);
```

Comment:
Its better to access the Camera via the ICamera in the Feature.

Reply:
Also its not
var worldPos
its
var screenPos

Reply:
You are right sir